server/models/contests/getContest.js

```diff
@@ -0,0 +1,33 @@
+const { pool } = require('../../models/database')
+
+/**
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {Object} param0.params
+ * @return {Promise}
+ */
+
+function getContest({ username, params }) {
+  return new Promise((resolve, reject) => {
+    const { contest_id: contestId } = params
+    pool.query(
+      `SELECT * FROM contests WHERE id=? 
+      AND (public=1 OR EXISTS(SELECT 1 FROM contests_groups cg INNER JOIN user_groups ug ON cg.group_id = ug.group_id 
+      WHERE ug.username=? AND cg.contest_id=?))`,
```

Comment:
```suggestion
      WHERE ug.username=? AND cg.contest_id=?)))`,
```

Reply:
why

Reply:
We can make this a new feature later. For now this it is not needed.
This was related to the previous change i suggested